src/praisonai-agents/praisonaiagents/bots/presentation.py

```
@@ -489,6 +557,26 @@ def _adapt_button(button: PresentationButton, limits: PresentationLimits) -> Pre
+
+    Produces ``reply:<value>`` so the inbound interactive registry can route
+    the chosen value back into the next agent turn. When the raw form exceeds
+    ``_MAX_CALLBACK_LEN`` (e.g. Telegram's 64-byte cap), the value is replaced
+    with a short, collision-resistant hash; channels then resolve the hash back
+    to the original value via the presentation's reply map (the wrapper keeps
+    that mapping when it renders an agent-authored presentation).
+    """
+    raw = f"{REPLY_CALLBACK_PREFIX}{value}"
+    if len(raw) <= _MAX_CALLBACK_LEN:
+        return raw
+    digest = hashlib.sha1(value.encode("utf-8")).hexdigest()[:16]
```

Ruff (0.15.18)

[error] Probable use of insecure hash functions in `hashlib`: `sha1`

(S324)